src/api/source/mvt.js

```diff
@@ -0,0 +1,179 @@
+import * as rsys from '../../client/rsys';
+import Dataframe from '../../core/dataframe';
+import * as Protobuf from 'pbf';
+import { VectorTile } from '@mapbox/vector-tile';
+import { decodeLines, decodePolygons } from '../../client/mvt/feature-decoder';
+import TileClient from './TileClient';
+import Base from './base';
+import { RTT_WIDTH } from '../../core/renderer';
+
+// Constants for '@mapbox/vector-tile' geometry types, from https://github.com/mapbox/vector-tile-js/blob/v1.3.0/lib/vectortilefeature.js#L39
+const mvtDecoderGeomTypes = { point: 1, line: 2, polygon: 3 };
+
+const geometryTypes = {
+    UNKNOWN: 'unknown',
+    POINT: 'point',
+    LINE: 'line',
+    POLYGON: 'polygon'
+};
+
+export default class MVT extends Base {
+
+    /**
+     * Create a carto.source.MVT.
+     *
+     * @param {object} data - A MVT data object
+     * @param {object} metadata - A carto.source.mvt.Metadata object
+     *
+     * @example
+     * const metadata = new carto.source.mvt.Metadata([{ type: 'number', name: 'total_pop'}])
+     * new carto.source.MVT("https://{server}/{z}/{x}/{y}.mvt", metadata);
+     *
+     * @fires CartoError
+     *
+     * @constructor MVT
+     * @extends carto.source.Base
+     * @memberof carto.source
+     * @IGNOREapi
+     */
+    constructor(templateURL, metadata = { columns: [] }) {
+        super();
+        this._templateURL = templateURL;
+        this._metadata = metadata;
+        this._tileClient = new TileClient(templateURL);
+        this._categoryStringToIDMap = {};
+        this._numCategories = 0;
+    }
+
+    _clone() {
+        return new MVT(this._templateURL, JSON.parse(JSON.stringify(this._metadata)));
+    }
+
+    bindLayer(addDataframe, dataLoadedCallback) {
+        this._tileClient.bindLayer(addDataframe, dataLoadedCallback);
+    }
+
+    async requestMetadata() {
+        return this._metadata;
+    }
+
+    requestData(viewport) {
+        return this._tileClient.requestData(viewport, this.responseToDataframeTransformer.bind(this));
+    }
+
+    async responseToDataframeTransformer(response, x, y, z) {
+        const mvt_extent = 4096;
```

Comment:
constant values, as numbers or strings, should be in capital letters

Reply:
Changed